src/client.rs

```diff
@@ -164,6 +164,199 @@ impl Client {
         self.issue(Cmd::Persist).await?;
         Ok(())
     }
+
+    /// Create a new raw request builder.
+    ///
+    /// This method allows to build a direct HTTP request to a remote site without routing
+    /// through the WebDriver host. It preserves the cookies and user agent from the current
+    /// WebDriver session, enabling you to maintain the session context while making external
+    /// requests. 
+    /// 
+    /// This can be useful for operations where direct access is needed or when
+    /// interacting with third-party services that require the same session cookies.
+    pub fn raw_request(&self) -> RawRequestBuilder<'_> {
+        RawRequestBuilder::new(self)
+    }
+}
+
+/// A builder for constructing raw HTTP requests with optional cookies.
+pub struct RawRequestBuilder<'a> {
+    client: &'a Client,
+    method: Method,
+    url: String,
+    cookie_url: Option<String>,
+    request_modifier: Option<Box<dyn FnOnce(http::request::Builder) -> hyper::Request<BoxBody<hyper::body::Bytes, Infallible>> + Send>>,
+}
+
+impl std::fmt::Debug for RawRequestBuilder<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("RawRequestBuilder")
+            .field("client", &self.client)
+            .field("method", &self.method)
+            .field("url", &self.url)
+            .field("cookie_url", &self.cookie_url)
+            .field("request_modifier", &"<closure>")
+            .finish()
+    }
+}
+
+impl<'a> RawRequestBuilder<'a> {
+    /// Create a new raw request builder.
+    pub fn new(client: &'a Client) -> Self {
+        RawRequestBuilder {
+            client,
+            method: Method::GET,
+            url: String::new(),
+            cookie_url: None,
```

Comment:
This should probably default to `"/please_give_me_your_cookies"` to maintain parity with the current implementation. And then maybe we add a way to opt _out_ of the cookie navigation?

Reply:
I've added an option, we can now choose to skip the cookie navigation process.

Reply:
I think you could probably get rid of the `bool` and instead have `skip_cookie_navigation` just set `.cookie_url = None`, no? That way you also avoid the extra nested conditional you now have in `send()`.

Reply:
Yes, the extra `bool` isn't needed. Setting `cookie_url` to `None` works directly to `skip`.